front/pages/w/[wId]/members/index.tsx

```diff
@@ -129,343 +392,257 @@ export default function WorkspaceAdmin({
     if (!res.ok) {
       window.alert("Failed to invite new member to workspace.");
     } else {
+      setSuccessMessage(
+        `Invite sent to ${inviteEmail}. You can repeat the operation to invite other users.`
+      );
       await mutate(`/api/w/${owner.sId}/invitations`);
     }
     setIsSending(false);
     setInviteEmail("");
-  };
+  }
+}
 
-  const handleRevokeInvitation = async (invitationId: number) => {
-    const res = await fetch(`/api/w/${owner.sId}/invitations/${invitationId}`, {
+function InviteSettingsModal({
+  showModal,
+  onClose,
+  owner,
+}: {
+  showModal: boolean;
+  onClose: () => void;
+  owner: WorkspaceType;
+}) {
+  const [domainUpdating, setDomainUpdating] = useState(false);
+  const [domainInput, setDomainInput] = useState(owner.allowedDomain || "");
+  const [allowedDomainError, setAllowedDomainError] = useState("");
+  return (
+    <Modal
+      isOpen={showModal}
+      onClose={onClose}
+      hasChanged={
+        domainInput !== owner.allowedDomain &&
+        !allowedDomainError &&
+        !domainUpdating
+      }
+      title="Invitation link settings"
+      type="right-side"
+      onSave={() => validDomain() && handleUpdateWorkspace()}
+    >
+      <div className="mt-6 flex flex-col gap-6 px-2">
+        <div>
+          Any person with a Google Workspace email on corresponding domain name
+          will be allowed to join the workspace.
+        </div>
+        <div className="flex flex-col gap-1.5">
+          <div className="font-bold">Whitelisted email domain</div>
+          <Input
+            className="text-sm"
+            placeholder={"Company domain"}
+            value={domainInput}
+            name={""}
+            error={allowedDomainError}
+            showErrorLabel={true}
+            onChange={(e) => {
+              setDomainInput(e);
+              setAllowedDomainError("");
+            }}
+            disabled={domainUpdating}
+          />
+        </div>
+      </div>
+    </Modal>
+  );
+
+  async function handleUpdateWorkspace() {
+    setDomainUpdating(true);
+    const res = await fetch(`/api/w/${owner.sId}`, {
       method: "POST",
       headers: {
         "Content-Type": "application/json",
       },
       body: JSON.stringify({
-        status: "revoked",
+        allowedDomain: domainInput,
       }),
     });
     if (!res.ok) {
-      window.alert("Failed to revoke member's invitation.");
+      window.alert("Failed to update workspace.");
+      setDomainUpdating(false);
     } else {
-      await mutate(`/api/w/${owner.sId}/invitations`);
+      // We perform a full refresh so that the Workspace name updates and we get a fresh owner
+      // object so that the formValidation logic keeps working.
+      window.location.reload();
+    }
+  }
+  function validDomain() {
```

Comment:
NIT: add types for all your functions too 🙏🏻 ?

Reply:
here OTOH I agree since you can't infer the return type without reading the function